lib/tasks/rake_helpers/storage.rb

```diff
@@ -0,0 +1,128 @@
+      # Insert reference information for attachments into ActiveStorage::Blob
+      # Set the asset key correctly in ActiveStorage::Blob
+      key = self.s3_path_pattern(model).split('/')
+        .map do |part|
+          case part
+          when ':id_partition'
+            "TO_CHAR(id, 'fm000/000/000/')"
+          when ':filename'
+            "#{name}_file_name"
+          else
+            "'#{part}/'"
+          end
+        end.join(', ')
+
+      connection.prepare('create_active_storage_blobs', <<~SQL)
+        INSERT INTO active_storage_blobs (key, filename, content_type, metadata, byte_size, checksum, created_at)
+          SELECT CONCAT(#{key}),
+                 #{name}_file_name,
+                 #{name}_content_type,
+                 CONCAT('in-progress/', CAST(id AS CHARACTER VARYING)),
+                 #{name}_file_size,
+                 as_#{name}_checksum,
+                 #{name}_updated_at
+            FROM #{self.models(model).table_name}
+            WHERE #{name}_file_name IS NOT NULL
+              AND id NOT IN (SELECT DISTINCT record_id FROM active_storage_attachments WHERE record_type = '#{self.models(model)}')
+          ON CONFLICT DO NOTHING;
+      SQL
+
+      # Link ActiveStorage::Blob objects to the correct records with ActiveStorage::Attachment
+      connection.prepare('create_active_storage_attachments', <<~SQL)
+        INSERT INTO active_storage_attachments (name, record_type, record_id, blob_id, created_at)
+          SELECT '#{name}',
+                 '#{self.models(model)}',
+                 CAST(SPLIT_PART(metadata, '/', 2) AS INTEGER),
+                 id,
+                 created_at
+          FROM active_storage_blobs
+          WHERE metadata LIKE 'in-progress/%';
+      SQL
+
+      connection.prepare('update_active_storage_blobs_metadata', <<~SQL)
+        UPDATE active_storage_blobs SET metadata='{}' WHERE metadata LIKE 'in-progress/%';
+      SQL
+
+      connection.exec_prepared('create_active_storage_blobs');
+      connection.exec_prepared('create_active_storage_attachments');
+      connection.exec_prepared('update_active_storage_blobs_metadata');
+    end
+  end
+
+  def self.make_dummy_files_for(model)
+    if self.models(model).nil?
+      puts "Cannot create dummy files for: #{model}"
+      exit
+    end
+
+    ATTACHMENTS[model].each do |name|
+      records = self.models(model).where.not("#{name}_file_name" => nil)
+
+      bar = self.progress_bar title: name, total: records.count
+
+      records.each do |record|
+        bar.increment
+        filename = File.absolute_path(record.send(name).path)
+        FileUtils.mkdir_p File.dirname(filename)
+        File.open(filename, 'wb') { |file| file.write(SecureRandom.random_bytes(record.send("#{name}_file_size"))) }
+      end
+    end
+  end
+
+  def self.set_checksums(records:, model:)
+    bar = self.progress_bar title: ATTACHMENTS[model].join(', '), total: records.count
+
+    records.each do |record|
```

Comment:
could  be something to look at testing if this work is on a separate branch, but it might be possible to use batches to speed this up?!